### PR TITLE
Add: GetInterface for rpc_service.go

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by AlexStocks** [comment]
serviceMap   map[string]map[string]*Service // protocol -> service name -> servicefirst of all,

**Comment by fangyincheng** [comment]
> serviceMap map[string]map[string]*Service // protocol -> service name -> service
> 
> first of all, I do not think this type of structure is a good idea. maybe `protocol + servcie name` as the map key is better.

We will get easyly service number of a protocol through this way, and I think `protocol + service`  is inappropriate.

File: common/rpc_service.go [hunk 1]
```diff
@@ -163,10 +165,23 @@ func (sm *serviceMap) GetService(protocol, name string) *Service {
...
+func (sm *serviceMap) GetInterface(interfaceName string) []*Service {
+	sm.mutex.RLock()
+	defer sm.mutex.RUnlock()
+	if s, ok := sm.interfaceMap[interfaceName]; ok {
+		return s
+	}
+	return nil
+}
+
+func (sm *serviceMap) Register(interfaceName, protocol string, rcvr RPCService) (string, error) {
```
Comment (zouyx) [inline common/rpc_service.go]: Add some comment?

File: common/rpc_service.go [hunk 2]
```diff
@@ -201,12 +216,13 @@ func (sm *serviceMap) Register(protocol string, rcvr RPCService) (string, error)
...
 	sm.mutex.Lock()
 	sm.serviceMap[protocol][s.name] = s
+	sm.interfaceMap[interfaceName] = append(sm.interfaceMap[interfaceName], s)
 	sm.mutex.Unlock()
 
 	return strings.TrimSuffix(methods, ","), nil
 }
 
-func (sm *serviceMap) UnRegister(protocol, serviceId string) error {
+func (sm *serviceMap) UnRegister(interfaceName, protocol, serviceId string) error {
```
Comment (zouyx) [inline common/rpc_service.go]: as above

File: common/rpc_service.go [hunk 3]
```diff
@@ -216,15 +232,32 @@ func (sm *serviceMap) UnRegister(protocol, serviceId string) error {
...
+		sm.mutex.RUnlock()
+		return perrors.New("no service for " + interfaceName)
+	}
+	index := -1
+	for i, svr := range svrs {
+		if svr == s {
+			index = i
+		}
+	}
 	sm.mutex.RUnlock()
```
Comment (AlexStocks) [inline common/rpc_service.go]: this code block should be in a lambda func as follows
func() {
lock()
defer unlock()

}()

**Comment by AlexStocks** [comment]
LGTM

**Pull request merged**